main.tf

```diff
@@ -1 +1,108 @@
+  iam_role_external_id = module.lacework_s3_iam_role.created ? module.lacework_s3_iam_role.external_id : var.iam_role_external_id
+  iam_role_name = var.use_existing_iam_role ? var.iam_role_name : (
+    length(var.iam_role_name) > 0 ? var.iam_role_name : "${var.prefix}-iam-${random_id.uniq.hex}"
+  )
+  mfa_delete = var.bucket_enable_versioning && var.bucket_enable_mfa_delete
+}
+
+resource "random_id" "uniq" {
+  byte_length = 4
+}
+
+module "lacework_s3_iam_role" {
+  source                  = "lacework/iam-role/aws"
+  version                 = "~> 0.1"
+  create                  = var.use_existing_iam_role ? false : true
+  iam_role_name           = local.iam_role_name
+  lacework_aws_account_id = var.lacework_aws_account_id
+  external_id_length      = var.external_id_length
+  tags                    = var.tags
+}
+
+resource "aws_iam_policy" "cross_account_s3_write_access" {
+  name        = local.cross_account_policy_name
+  description = "A cross account policy to allow Lacework to write to S3 for Data Export"
+  policy      = data.aws_iam_policy_document.cross_account_s3_write_access.json
+}
+
+resource "aws_iam_role_policy_attachment" "cross_account_s3_write_access" {
+  role       = local.iam_role_name
+  policy_arn = aws_iam_policy.cross_account_s3_write_access.arn
+  depends_on = [module.lacework_s3_iam_role]
+}
+
+data "aws_iam_policy_document" "cross_account_s3_write_access" {
+  version = "2012-10-17"
+
+  statement {
+    sid       = "ListBucket"
+    resources = [local.bucket_arn]
+    actions   = ["s3:ListBucket"]
+  }
+
+  statement {
+    sid       = "PutObjectsInBucket"
+    actions   = ["s3:GetObject", "s3:PutObject"]
+    resources = ["${local.bucket_arn}/*"]
+  }
+
+  dynamic "statement" {
+    for_each = var.bucket_enable_encryption == true ? (var.bucket_sse_algorithm == "aws:kms" ? [1] : []) : []
+    content {
+      sid       = "EncryptFiles"
+      actions   = ["kms:Encrypt", "kms:Decrypt"]
+      resources = [var.bucket_sse_key_arn]
+    }
+  }
+}
+
+resource "aws_s3_bucket" "s3_data_export_bucket" {
+  count         = var.use_existing_s3_bucket ? 0 : 1
+  bucket        = local.bucket_name
+  force_destroy = var.bucket_force_destroy
+
+  versioning {
+    enabled    = var.bucket_enable_versioning
+    mfa_delete = local.mfa_delete
+  }
+
+  dynamic "server_side_encryption_configuration" {
+    for_each = var.bucket_enable_encryption == true ? [1] : []
+    content {
+      rule {
+        apply_server_side_encryption_by_default {
+          kms_master_key_id = var.bucket_sse_key_arn
+          sse_algorithm     = var.bucket_sse_algorithm
+        }
+      }
+    }
+  }
+
+  tags = var.tags
+}
+
+# wait for X seconds for things to settle down in the AWS side
+# before trying to create the Lacework external integration
+resource "time_sleep" "wait_time" {
+  create_duration = var.wait_time
+  depends_on = [
+    aws_iam_role_policy_attachment.cross_account_s3_write_access
```

Comment:
Not to myself: I think you don't explicitly add a dependency to the module `lacework_s3_iam_role`since the policy attachment already depends on it, correct?

Reply:
Correct, the permissions propagation seems to be what takes a bit of time, so wanted to wait specifically on that.  Also, following in the footsteps of the master: https://github.com/lacework/terraform-aws-cloudtrail/blob/fb42d897b8303c006552ecbf7a7bd108ca6be541/main.tf#L310